examples/specific/userdefined.h

```diff
@@ -1,7 +1,7 @@
 // Example from Doxygen documentation
 
-/** A class. More details about the Test class */
-class Test
+/** A class. More details about the UserDefinedGroupTest class */
+class UserDefinedGroupTest
```

Comment:
I assume things like this are renamed as they are clashing between projects in the output? So you've created unique names for them?

Reply:
Its not a fully transparent demonstration; I didn't add the `no-link` option to what the user sees in the sample rst snippets - only to the rendered breathe directives that were problematic.

Additionally, if users inspect the page's ToC when `no-link` is used, then they'll notice a gap in the what is indexed in the page's ToC. This is because `no-link` prevents an entry into the ToC for the rendered items.

Reply:
Makes sense. Thank you for explaining it.

Reply:
Yes, I only used no-link to avoid duplicate IDs on pages that keep documenting the same API to demonstrate the breathe directive's various options. For the most part, I think the `.. cpp:namespace::` tactic is working, but it doesn't in some instances.

Reply:
@jakobandersen I imagine adding a new `:no-index:` option to use the newer `noindexentry` option would require breathe to be pinned to sphinx v3.2.0. Is this what you're suggesting? Or were you clarifying `:no-link:` for us?

I was having a hard time completely understanding the discussion you linked to, but the linked sphinx doc helped clear that up for me.

Reply:
It was mostly an initial attempt at clarification, but honestly I don't know yet what the best course of action is with it. There are some intricacies of the Doxygen-based IDs I don't have an overview of yet. And that comes on top of the discussion I linked.
As for this PR I am mostly concerned with this ``cpp:namespace`` tactic. Unfortunately I don't have much time before next weekend, but generally I think we should not make a hax just to prevent warnings. We should use the intended directives, and then we should fix the underlying issue with the warnings.

Reply:
Honestly, it is a separate issue. And it has gone unnoticed because the doc's builds were ignoring any warnings. I could raise a new issue to describe it in as much detail as possible (including a sample to demonstrate).